packages/framework/src/Model/Product/Search/Export/ProductSearchExporter.php

```diff
@@ -153,6 +153,41 @@ public function exportWithOutput(int $domainId, string $locale, SymfonyStyle $sy
         $this->sqlLoggerFacade->reenableLogging();
     }
 
+    /**
+     * @param int $domainId
+     * @param string $locale
+     * @param int[] $productIds
+     */
+    public function exportIds(int $domainId, string $locale, array $productIds): void
+    {
+        $exportedIds = $this->exportBatchIds($domainId, $locale, $productIds);
```

Comment:
It took me some time to figure out how this works and why are only some IDs exported. I would move logic with loading and deciding what should be deleted here. 

You load data by ids, send them to exportBatch, convert data to ids and decide what should be deleted.

Reply:
Yeah, the thing is, I get some IDs, and I don't know if they need to be deleted or exported. 
So I export the IDs, from database I get list of IDs that was found and are valid to export to Elasticsearch and rest of the IDs have to be deleted from Elastic to prevent having old data in Elastic.

So you suggest to merge these two methods (`exportIds` and `exportBatchIds`)?

Reply:
I would move the first part (loading data and extracting IDs) to exportIds and rest keep there and rename the method to exportBatch